app.py

```diff
@@ -0,0 +1,57 @@
+import gradio as gr
+import openai
+import json
+from factool import Factool
+import os
+
+
+def chat_with_gpt(api_key, model, message):
+    openai.api_key = api_key
+    response = openai.ChatCompletion.create(
+      model=model,
+      messages=[
+            {"role": "system", "content": "You are a helpful assistant."},
+            {"role": "user", "content": message},
+        ]
+    )
+    return response.choices[0].message['content']
+
+def fact_check(openai_api_key, serper_api_key, scraper_api_key, model, message, response, category):
+    os.environ['SCRAPER_API_KEY'] = ''
```

Comment:
This type of implementation is kinda strange. Let's create an issue for future improvement.

Reply:
ok, I will open an issue.